compiler/semstmts.nim

```diff
@@ -455,6 +455,13 @@ proc semVarOrLet(c: PContext, n: PNode, symkind: TSymKind): PNode =
         if hasEmpty(typ):
           localError(def.info, errCannotInferTypeOfTheLiteral,
                      ($typ.kind).substr(2).toLower)
+        elif typ.kind == tyTuple and def.kind == nkConv:
```

Comment:
wrong position, it has nothing to do with the 'var' context, it's the type conversion missing these checks.